src/compiler/crystal/macros/methods.cr

```diff
@@ -400,6 +400,24 @@ module Crystal
         interpret_check_args { symbolize }
       when "class_name"
         interpret_check_args { class_name }
+      when "docs"
+        interpret_check_args(named_params: ["comment"]) do
+          is_comment = if named_args && (comment_arg = named_args["comment"]?)
+                         comment_arg.raise "named argument 'comment' to ASTNode#docs must be a BoolLiteral, not #{comment_arg.class_desc}" unless comment_arg.is_a?(BoolLiteral)
+
+                         comment_arg
+                       else
+                         BoolLiteral.new false
+                       end
+
+          docs = self.doc || ""
+
+          if is_comment.value
+            docs = docs.gsub("\n", "\n# ")
```

Comment:
Will this work with different levels of indentation?

Reply:
Yeah, the parser doesn't care about indentation.

Reply:
I see, that's good, the doc will have mixed indentation though? I'm curious how versatile this will be.

Reply:
The parser strips any whitespace before the comment delimiter (`#`). So the parsed doc string looks the same, regardless of indent.

Reply:
Yeah, but the doc indentation won't match the previous level (unless it's on the top level)